http/src/main/scala/org/apache/pekko/http/scaladsl/server/directives/FileUploadDirectives.scala

```diff
@@ -175,9 +176,12 @@ trait FileUploadDirectives {
       implicit val ec = ctx.executionContext
 
       def tempDest(fileInfo: FileInfo): File = {
-        val dest = File.createTempFile("pekko-http-upload", ".tmp")
-        dest.deleteOnExit()
-        dest
+        val dest = Files.createTempFile("pekko-http-upload", ".tmp")
+        Runtime.getRuntime.addShutdownHook(new Thread() {
```

Comment:
Will there have many hook threads?

Reply:
> Will there have many hook threads?

I don't think so but note that this is the same implementation as `deleteOnExit` so it should behave the same as current version.